### PR TITLE
ResourceQuota Enhancement

### PR DESCRIPTION
**Pull request by chrisho**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
https://github.com/harvester/harvester/issues/3124

**Solution:**
https://github.com/chrisho/harvester/blob/maintenance-resource-hep/enhancements/20230228-resource-quota-enhancement.md

**Related Issue:**
https://github.com/harvester/harvester/issues/3124

**Test plan:**
https://github.com/chrisho/harvester/blob/maintenance-resource-hep/enhancements/20230228-resource-quota-enhancement.md#test-plan

### CODE REVIEW

File: pkg/api/vm/handler.go [hunk 1]
```diff
@@ -976,6 +990,80 @@ func (h *vmActionHandler) cloneVolumes(newVM *kubevirtv1.VirtualMachine) ([]core
 	return newPVCs, secretNameMap, nil
 }
 
+//func (h *vmActionHandler) checkMaintenceAvailableResource(namespace string, vmi *kubevirtv1.VirtualMachineInstance) error {
```
Comment (futuretea) [inline pkg/api/vm/handler.go]: ditto

File: pkg/controller/master/setup.go [hunk 2]
```diff
@@ -46,6 +47,7 @@ var registerFuncs = []registerFunc{
 	upgradelog.Register,
 	addon.Register,
 	storagenetwork.Register,
+	//namespace.Register,
```
Comment (futuretea) [inline pkg/controller/master/setup.go]: ditto

File: pkg/api/vm/handler.go [hunk 3]
```diff
@@ -340,6 +349,11 @@ func (h *vmActionHandler) migrate(ctx context.Context, namespace, vmName string,
 			VMIName: vmName,
 		},
 	}
+
+	//if err := h.checkMaintenceAvailableResource(namespace, vmi); err != nil {
```
Comment (futuretea) [inline pkg/api/vm/handler.go]: remove unused code

**Comment by chrisho** [comment]
Re-design

**Pull request closed without merging**